File: primes.py

import math

upper_limit = 1000000

def is_prime(n):
  divisor = 2
  sqrt_n = math.sqrt(n)
  while (divisor < sqrt_n):
    if (n % divisor == 0):
      return False

    divisor += 1

  return True


for i in range (3, upper_limit):
  if (is_prime(i)):
    print (i)

File: primes.rs

fn main() {
  let upper_limit: int = 1000000;

  for i in range(2i, upper_limit) {
    if is_prime(i) {
      println!("{}", i);
    }
  }
}

fn is_prime(n: int) -> bool {
  let mut divisor = 2;

  while divisor < n {
    if n % divisor == 0 {
      return false;
    }
    divisor += 1;
  }

  return true;
}
